scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -0,0 +1,478 @@
+            self.__k8s_namespaces_to_ignore.append(x.strip())
+
+        default_rotation_count, default_max_bytes = self._get_log_rotation_configuration()
+
+        self.__event_object_filter = self._config.get( 'event_object_filter', [  'CronJob', 'DaemonSet', 'Deployment', 'Job', 'Node', 'Pod', 'ReplicaSet', 'ReplicationController', 'StatefulSet' ] )
+
+        self.__max_log_size = self._config.get( 'max_log_size' )
+        if self.__max_log_size is None:
+            self.__max_log_size = default_max_bytes
+
+        self.__max_log_rotations = self._config.get( 'max_log_rotations' )
+        if self.__max_log_rotations is None:
+            self.__max_log_rotations = default_rotation_count
+
+
+    def open_metric_log( self ):
+        """Override open_metric_log to prevent a metric log from being created for the Kubernetes Events Monitor
+        and instead create our own logger which will log raw messages out to disk.
+        """
+        self.__disk_logger = logging.getLogger( self.__message_log )
+
+        #assume successful for when the logger handler has already been created
+        success = True
+
+        #only configure once -- assumes all configuration happens on the same thread
+        if len( self.__disk_logger.handlers ) == 0:
+            #logger handler hasn't been created yet, so assume unsuccssful
+            success = False
+            try:
+                self.__log_handler = AutoFlushingRotatingFileHandler( filename = self.log_config['path'],
+                                                                      maxBytes = self.__max_log_size,
+                                                                      backupCount = self.__max_log_rotations,
+                                                                      flushDelay = self.__flush_delay)
+
+                formatter = logging.Formatter()
+                self.__log_handler.setFormatter( formatter )
+                self.__disk_logger.addHandler( self.__log_handler )
+                self.__disk_logger.setLevel( logging.INFO )
+                self.__disk_logger.propagate = False
+                success = True
+            except Exception, e:
+                global_log.error( "Unable to open KubernetesEventsMonitor log file: %s" % str( e ) )
+
+        return success
+
+    def close_metric_log( self ):
+        if self.__log_handler:
+            self.__disk_logger.removeHandler( self.__log_handler )
+            self.__log_handler.close()
+
+    def set_log_watcher( self, log_watcher ):
+        self.__log_watcher = log_watcher
+
+    def _check_if_alive( self, k8s, node ):
+        """
+            Checks to see if the node specified by `node` is alive
+        """
+        if node is None:
+            return False
+
+        result = {}
+        try:
+            # this call will throw an exception on failure
+            result = k8s.query_api( '/api/v1/nodes/%s' % node )
```

Comment:
It doesn't guarantee that pods can be scheduled on that node.  The customer might have configured a node with various taints to prevent pods running on it.  I think this will come down to the problem we discussed earlier about whether or not the agent is running on a particular node, and the solution being to manually specify the event leader node in that case.

The node could also potentially be shutting down (so no more scheduling allowed) in which case it will be picked up the next iteration through the loop so it's probably ok.